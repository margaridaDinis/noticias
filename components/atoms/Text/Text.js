import React from 'react';
import PropTypes from 'prop-types';
import { Text as Txt } from 'react-native';

import styles from './Text.style';

const Text = ({
  children,
  tag = 'p',
  paddingBottom,
  color,
  bold,
  style,
  ...rest
}) => {
  const textStyles = [styles.general, styles[tag]];

  if (bold) textStyles.push(styles.bold);
  if (color) textStyles.push({ color });
  if (paddingBottom) textStyles.push({ paddingBottom });
  if (style) textStyles.push(style);

  return (
    <Txt style={textStyles} accessibilityRole="text" {...rest}>
      {children}
    </Txt>
  );
};

Text.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.array,
  ]),
  tag: PropTypes.string,
  bold: PropTypes.bool,
  color: PropTypes.string,
  paddingBottom: PropTypes.number,
  style: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
};

export default Text;
