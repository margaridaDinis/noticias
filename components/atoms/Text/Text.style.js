import { StyleSheet } from 'react-native';
import { BASE_LINE_HEIGHT, FONT_SIZE } from '../../../constants/Layout';

const styles = StyleSheet.create({
  general: {
    fontWeight: 'bold',
  },
  p: {
    fontSize: FONT_SIZE.base,
    lineHeight: BASE_LINE_HEIGHT,
    fontWeight: 'normal',
  },
  a: {
    textDecorationLine: 'underline',
    fontWeight: 'normal'
  },
  small: {
    fontSize: FONT_SIZE.small,
    lineHeight: BASE_LINE_HEIGHT / 1.2,
    fontWeight: 'normal',
  },
  h1: {
    fontSize: FONT_SIZE.h1 // 28
  },
  h2: {
    fontSize: FONT_SIZE.h2 // 24
  },
  h3: {
    fontSize: FONT_SIZE.h3 // 20
  },
  h4: {
    fontSize: FONT_SIZE.h4 // 18
  },
  h5: {
    fontSize: FONT_SIZE.h5 // 16
  },
  h6: {
    fontSize: FONT_SIZE.h6 // 14
  },
  bold: {
    fontWeight: 'bold'
  }
});

export default styles;
