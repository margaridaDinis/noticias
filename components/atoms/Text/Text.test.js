import React from 'react';
import { render } from '@testing-library/react-native';
import Text from './Text';

describe('<Text />', () => {
  it('renders without props', () => {
    const { baseElement } = render(<Text>Text</Text>);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with different tag', () => {
    const { queryByRole } = render(<Text tag="h1">Text</Text>);

    expect(queryByRole('text').props.style).toMatchSnapshot();
  });

  it('renders with as bold', () => {
    const { queryByRole } = render(<Text bold>Text</Text>);

    expect(queryByRole('text').props.style).toMatchSnapshot();
  });

  it('renders with custom padding bottom', () => {
    const { queryByRole } = render(<Text paddingBottom={16}>Text</Text>);

    expect(queryByRole('text').props.style).toMatchSnapshot();
  });

  it('renders with custom style', () => {
    const { queryByRole } = render(
      <Text style={{ textTransform: 'uppercase' }}>Text</Text>,
    );

    expect(queryByRole('text').props.style).toMatchSnapshot();
  });
});
