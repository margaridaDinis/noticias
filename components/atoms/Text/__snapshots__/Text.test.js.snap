// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders with as bold 1`] = `
Array [
  Object {
    "fontWeight": "bold",
  },
  Object {
    "fontSize": 16,
    "fontWeight": "normal",
    "lineHeight": 24,
  },
  Object {
    "fontWeight": "bold",
  },
]
`;

exports[`<Text /> renders with custom padding bottom 1`] = `
Array [
  Object {
    "fontWeight": "bold",
  },
  Object {
    "fontSize": 16,
    "fontWeight": "normal",
    "lineHeight": 24,
  },
  Object {
    "paddingBottom": 16,
  },
]
`;

exports[`<Text /> renders with custom style 1`] = `
Array [
  Object {
    "fontWeight": "bold",
  },
  Object {
    "fontSize": 16,
    "fontWeight": "normal",
    "lineHeight": 24,
  },
  Object {
    "textTransform": "uppercase",
  },
]
`;

exports[`<Text /> renders with different tag 1`] = `
Array [
  Object {
    "fontWeight": "bold",
  },
  Object {
    "fontSize": 28,
  },
]
`;

exports[`<Text /> renders without props 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        Array [
          Object {
            "fontWeight": "bold",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "normal",
            "lineHeight": 24,
          },
        ]
      }
    >
      Text
    </Text>
  </View>
</View>
`;
